.pre-commit-config.yaml

```
@@ -3,6 +3,13 @@ default_install_hook_types:
   - commit-msg
 
 repos:
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.5.6
+    hooks:
+      - id: ruff
+        args: [ --fix, --show-fixes ]
+      - id: ruff-format
```

In the pre-commit hook i would add only the checks, not the fixes. Reasoning: Format changes done when the hook is executed (assuming exit code of ruff is 0 (success)) will not be part of the commit itself, thus requiring a new `--fixup` commit by the user. What i am trying to say is that the hook execution should fail and let the user fix errors manually, obliging them to add the changes and commit again.

Taking above into account and after having changed to latest version from main (above comment) you will probably need below syntax (as is in our Makefile).

```suggestion
      - id: ruff
        args: [ check ]
      - id: ruff
        args: [ format, --check ]
```

---

the correct will be

```
      - id: ruff
        args: [ --check ]
      - id: ruff-format
        args: [ --check ]
```
        
        right?

---

```
      - id: ruff
        args: [ --check ]
      - id: ruff-format
        args: [ --check ]
```